src/CommonITILObject.php

```diff
@@ -8620,20 +8620,6 @@ protected function updateActors(bool $disable_notifications = false)
 
                 $found = false;
                 foreach ($existings as $existing) {
-                    if (
-                        $actor['itemtype'] === User::class
-                        && $actor['items_id'] == 0
-                        && $actor['itemtype'] == $existing['itemtype']
-                    ) {
-                        // "email" actor found
-                        if ($actor['alternative_email'] == $existing['alternative_email']) {
-                            $found = true;
-                            break;
-                        }
-                        // Do not check for modifications on "email" actors (they should be deleted then re-added on email change)
-                        continue;
-                    }
-
```

Comment:
Removing this whole piece of code result in unexpected behaviour when modifiying actors.

For instance:
1. Create a ticket.
2. Add `test1@domain.tld` in requester field and save the form.
3. Add `test2@domain.tld` in requester field and save the form.
Expected result: `test1@domain.tld` and `test2@domain.tld` are both present in requester fields.
Current result: only `test2@domain.tld` is present in requester fields.

Reply:
Good catch, we should probably add a test on this specific action to ensure it isn't broken in a future update.

Reply:
The deleted code has been reinstated with a slight modification that should solve the problem : https://github.com/glpi-project/glpi/pull/16070/commits/4e27a0d7970f917bef330706c616f0f495c05622